### PR TITLE
SALTO-825: added support for changing instances element ids

### PR DESCRIPTION
**Pull request by alonstern**
Added support for service id in data instances

---
_Release Notes_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/data_elements/data_elements.ts [hunk 1]
```diff
@@ -107,18 +108,32 @@ const createInstance = async (
       return castFieldValue(value, field)
     },
   })
-  const id = naclCase(fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]])
+  const defaultName = naclCase(fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]])
+  const name = elemIdGetter !== undefined ? elemIdGetter(
+    NETSUITE,
+    {
+      [ADAPTER]: NETSUITE,
+      [TYPE_TO_IDENTIFIER[type.elemID.name]]: fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]],
```
Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/data_elements.ts:116]: What is the `TYPE_TO_IDENTIFIER[type.elemID.name]` key? I think that there should be another key here no?

Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/data_elements.ts:116]: Maybe `fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]]` can be extracted to var?

**Review by omrilit — COMMENTED** [review]
I think that there are missing UTs?

File: packages/netsuite-adapter/src/data_elements/data_elements.ts (hunk 1 above)
Comment (alonstern) [reply packages/netsuite-adapter/src/data_elements/data_elements.ts:116]: I don't think so, in here it is the same too https://github.com/salto-io/salto/blob/master/packages/netsuite-adapter/src/transformer.ts#L51

Comment (omrilit) [reply packages/netsuite-adapter/src/data_elements/data_elements.ts:116]: You are correct, maybe extracting to `serviceIdFieldName` or something similar will help understanding it better?

File: packages/netsuite-adapter/src/data_elements/data_elements.ts [hunk 2]
```diff
@@ -107,18 +108,34 @@ const createInstance = async (
       return castFieldValue(value, field)
     },
   })
-  const id = naclCase(fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]])
+  const identifierValue = fixedRecord?.[TYPE_TO_IDENTIFIER[type.elemID.name]]
+  const defaultName = naclCase(identifierValue)
+  const serviceIdFieldName = TYPE_TO_IDENTIFIER[type.elemID.name]
```
Comment (omrilit) [inline packages/netsuite-adapter/src/data_elements/data_elements.ts:113]:
```suggestion
  const serviceIdFieldName = TYPE_TO_IDENTIFIER[type.elemID.name]
  const identifierValue = fixedRecord?.[serviceIdFieldName]
  const defaultName = naclCase(identifierValue)
```

**Review by omrilit — APPROVED** [review]
1 nit

**Pull request merged**